File: 프로그래머스/lv2/12939. 최댓값과 최솟값/최댓값과 최솟값.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
vector<int> v;

string solution(string s) {
    int len = s.size();
    bool flag = true;
    int temp = 0;
    int idx = 0;
    while(idx <= len){
        if(s[idx] == ' ' || idx == len){
            if(flag) v.push_back(temp);
            else v.push_back(-temp);
            temp = 0;
            flag = true;
        }
        else if(s[idx] == '-'){
            flag = false;
        }
        else{
            temp = 10 * temp + (s[idx] - '0');
        }    
        idx++;
    }
    int maxima = *max_element(v.begin(), v.end());
    int minima = *min_element(v.begin(), v.end());
    string answer = to_string(minima) + " " + to_string(maxima);
    return answer;
}

File: 프로그래머스/lv2/12939. 최댓값과 최솟값/최댓값과 최솟값.py
def solution(s):
    answer = ''
    maxima, minima = -1e9, 1e9
    flag = 0
    
    num = 0
    for temp in s:
        if temp == ' ': 
            maxima = max(maxima, num)
            minima = min(minima, num)
            num = 0
            flag = 0
            continue
        else:
            if temp == '-' : 
                flag = 1
            else:
                if flag: num = 10* num - int(temp)
                else: num = 10*num + int(temp)
            
    maxima = max(maxima, num)
    minima = min(minima, num)
    
    answer += str(minima)
    answer += ' '
    answer += str(maxima)
    
    return answer